src/main/java/unicash/model/transaction/predicates/TransactionContainsAllKeywordsPredicate.java

```diff
@@ -0,0 +1,130 @@
+ * all predicates in this list.
+ */
+public class TransactionContainsAllKeywordsPredicate implements Predicate<Transaction> {
+
+    private List<Predicate<Transaction>> predicateList;
+
+    public TransactionContainsAllKeywordsPredicate() {
+        this.predicateList = new ArrayList<>();
+    }
+
+    public TransactionContainsAllKeywordsPredicate(List<Predicate<Transaction>> predicateList) {
+        this.predicateList = predicateList;
+    }
+
+    @Override
+    public boolean test(Transaction transaction) {
+        if (predicateList.isEmpty()) {
+            return false;
+        }
+
+        return predicateList.stream()
+                .allMatch(predicate -> predicate.test(transaction));
+    }
+
+
+    /**
+     * Creates a new {@code TransactionNameContainsKeywordsPredicate} with the
+     * given name as a string and adds this to the encapsulated list of
+     * transaction predicates.
+     *
+     * @param name the name to be added as a transaction predicate keyword
+     */
+    public void addNameKeyword(String name) {
+        TransactionNameContainsKeywordsPredicate namePredicate =
+                new TransactionNameContainsKeywordsPredicate(toKeywordList(name));
+
+        predicateList.add(namePredicate);
+
+    }
+
+    /**
+     * Creates a new {@code TransactionLocationContainsKeywordsPredicate} with the
+     * given location as a string and adds this to the encapsulated list of
+     * transaction predicates.
+     *
+     * @param location the location to be added as a transaction predicate keyword
+     */
+    public void addLocationKeyword(String location) {
+        TransactionLocationContainsKeywordsPredicate locationPredicate =
+                new TransactionLocationContainsKeywordsPredicate(toKeywordList(location));
+
+        predicateList.add(locationPredicate);
+
+    }
+
+
+    /**
+     * Creates a new {@code TransactionCategoryContainsValuePredicate} with the
+     * given {@code Category} as a string and adds this to the encapsulated list of
+     * transaction predicates.
+     *
+     * @param category the {@code Category} to be added as a transaction predicate keyword
+     */
+    public void addCategoryKeyword(String category) {
+        TransactionCategoryContainsKeywordsPredicate categoryPredicate =
+                new TransactionCategoryContainsKeywordsPredicate(toKeywordList(category));
+
+        predicateList.add(categoryPredicate);
+
+    }
+
+
+    /**
+     * A helper method that returns the input string keyword as a list
+     * with a single item.
+     *
+     * @param keyword the input keyword String
+     * @return the keyword as a single List.
+     */
+    private static List<String> toKeywordList(String keyword) {
+        return Collections.singletonList(keyword.toLowerCase());
+    }
+
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
```

Comment:
Missed `equals` change?

Reply:
Missed that, fixed!

btw couldn't reply to the other comment above but I thought it was just a weekly milestone but in the tp grading breakdown they apparently use it to assess code quality, looks like it is an overall requirement so just to be safe, I'm personally not a fan of it either haha

<img width="501" alt="image" src="https://github.com/AY2324S1-CS2103-T16-3/tp/assets/19762596/8eb430a0-a562-4701-ab7a-5935e11c9d05">

Reply:
Gotcha, no worries at all, just kinda weird to see two different variations of null checking